### PR TITLE
Additional scanning

### PR DESCRIPTION
**Pull request by withinfocus**
## 🎟️ Tracking

Internal change.

## 📔 Objective

Sets up SARIF output for Checkmarx scans so we can see results in GitHub and adds SonarCloud.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -276,3 +277,19 @@ afterEvaluate {
...
         .forEach { it.enabled = false }
 }
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,app/src/debug/")
+        property("sonar.tests", "app/src/test/")
```
Comment (withinfocus) [inline app/build.gradle.kts:287]: ℹ️ Double-check these paths for me.

Comment (david-livefront) [reply app/build.gradle.kts:287]: The paths look correct but do we only want `main` and `debug`?

Comment (withinfocus) [reply app/build.gradle.kts:287]: Which others are appropriate? This came over from https://github.com/bitwarden/authenticator-android/tree/main/app/src that only has those two folders that aren't test code, but I can list out whatever else we want to include here that we consider "source" for the app itself.

**Review by SaintPatrck — APPROVED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (david-livefront) [reply app/build.gradle.kts]: The ones @SaintPatrck added are the only ones that have actual Kotlin code in them but wouldn't we want to coder all the types and variants, it is possible that code could be added to any of them in the future.

Comment (withinfocus) [reply app/build.gradle.kts]: @david-livefront I agree but am just implementing suggestions -- if you call out additional ones I can make a quick PR, or of course you can too -- we should shoot for as much coverage as possible.

Comment (withinfocus) [reply app/build.gradle.kts]: https://github.com/bitwarden/android/pull/3349 opened this up to be more generic.

**Pull request merged**